examples/riscv/family.py

```diff
@@ -0,0 +1,81 @@
+
+from peak import family
+
+from ast_tools.passes import remove_asserts
+
+
+# A bit of hack putting the def of word and idx here
+# and not isa but it makes life easier
+class _RiscFamily_mixin:
+    @property
+    def Word(self):
+        return self.BitVector[32]
+
+    @property
+    def Idx(self):
+        return self.BitVector[5]
+
+
+class PyFamily(_RiscFamily_mixin, family.PyFamily):
+    def get_register_file(fam_self):
```

Comment:
Why do you need an extra level of indirection here? why not just define RegisterFile and refer to 'Idx' and 'Word' via 'self'?

Reply:
Because I need to access both the outer class and the inner class.  If I don't have a closure (`get_register_file`) there is no way to access the outer class from the inner.